### DISCUSSION TITLE
How do I use MonitoringMode.Sampling?

### DISCUSSION THREAD

**Discussion by jverdevel** [eclipse-milo/milo#872] [General]
I'm trying to find out how to use Eclipse Milo, and finding out how subscriptions go. I can easily get any MonitoringMode.Reporting mode subscription to work, but when I use Sampling it doesn't call the callback method (as expected). According to the docs it's supposed to "queue" up the values without calling the callback, but I can't find any place I can access that queue or anything similar. The UaMonitoredItem doesn't have anything in its interface that looks like it, neither does the request.

It's probably something obvious, but what am I doing wrong?

Thank you in advance and sorry if it's somewhere and I haven't seen it...

**Comment by kevinherron** [comment]
Are you working with the client SDK or server SDK?

**Reply by jverdevel** [reply]
Client SDK, thank you!

**Reply by kevinherron** [reply]
MonitoringMode.Sampling simply means the server continues to sample the underlying but does not report the values to the client.

The queued values are not available to you. If you change back to MonitoringMode.Reporting then they would be the first values reported for that item.

**Reply by jverdevel** [reply]
Okay. Thank you! That explains it.

**Reply by cannot-find-object** [reply]
I what to ask another question，when subscription is running，the log show ‘milo-shared-thread-pool-1’，the 1 was beening increasing，when it increased to ‘1700’，the system will crashed

**Reply by cannot-find-object** [reply]
from  ‘milo-shared-thread-pool-1’ to  ‘milo-shared-thread-pool-1700’

**Reply by kevinherron** [reply]
@cannot-find-object please start a new discussion thread with more information and logs.
